oak/server/rust/oak_runtime/src/time/tests.rs

```diff
@@ -0,0 +1,209 @@
+        Interval { min: 2, max: 6 },
+        Interval { min: 1, max: 4 },
+    ];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(Ok(Interval { min: 3, max: 5 }), result);
+}
+
+#[test]
+fn test_valid_overlap_three_of_three_reversed() {
+    let client = RoughtimeClient::new(vec![], 3, 0, 0, 0);
+    let intervals = vec![
+        Interval { min: 3, max: 5 },
+        Interval { min: 2, max: 6 },
+        Interval { min: 1, max: 4 },
+    ];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(Ok(Interval { min: 3, max: 4 }), result);
+}
+
+#[test]
+fn test_valid_overlap_five_of_ten() {
+    let client = RoughtimeClient::new(vec![], 5, 0, 0, 0);
+    let intervals = vec![
+        Interval { min: 1, max: 2 },
+        Interval { min: 1, max: 9 },
+        Interval { min: 2, max: 9 },
+        Interval { min: 3, max: 9 },
+        Interval { min: 4, max: 9 },
+        Interval { min: 5, max: 9 },
+        Interval { min: 6, max: 9 },
+        Interval { min: 7, max: 9 },
+        Interval { min: 8, max: 9 },
+        Interval { min: 9, max: 9 },
+    ];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(Ok(Interval { min: 5, max: 9 }), result);
+}
+
+#[test]
+fn test_valid_overlap_point() {
+    let client = RoughtimeClient::new(vec![], 2, 0, 0, 0);
+    let intervals = vec![Interval { min: 4, max: 8 }, Interval { min: 1, max: 4 }];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(Ok(Interval { min: 4, max: 4 }), result);
+}
+
+#[test]
+fn test_invalid_overlap_four_of_three() {
+    let client = RoughtimeClient::new(vec![], 4, 0, 0, 0);
+    let intervals = vec![
+        Interval { min: 1, max: 4 },
+        Interval { min: 2, max: 6 },
+        Interval { min: 3, max: 5 },
+    ];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(
+        Err(RoughtimeError::NotEnoughOverlappingIntervals {
+            actual: 3,
+            expected: 4
+        }),
+        result
+    );
+}
+
+#[test]
+fn test_invalid_overlap_three_of_three() {
+    let client = RoughtimeClient::new(vec![], 3, 0, 0, 0);
+    let intervals = vec![
+        Interval { min: 1, max: 2 },
+        Interval { min: 3, max: 6 },
+        Interval { min: 3, max: 5 },
+    ];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(
+        Err(RoughtimeError::NotEnoughOverlappingIntervals {
+            actual: 2,
+            expected: 3
+        }),
+        result
+    );
+}
+
+#[test]
+fn test_invalid_overlap_inverted_intervals() {
+    let client = RoughtimeClient::new(vec![], 3, 0, 0, 0);
+    let intervals = vec![
+        Interval { min: 4, max: 1 },
+        Interval { min: 6, max: 2 },
+        Interval { min: 5, max: 3 },
+    ];
+    let result = client.find_overlap(&intervals);
+    assert_eq!(
+        Err(RoughtimeError::NotEnoughOverlappingIntervals {
+            actual: 0,
+            expected: 3
+        }),
+        result
+    );
+}
+
+#[test]
```

Comment:
This could perhaps be moved to a separate file for `integration_tests`? (Although I am not sure how much that can help with managing the execution of different test categories.)